packages/component-library/src/components/form/mt-text-editor/_internal/mt-text-editor-toolbar-button-color.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div class="mt-text-editor-toolbar-button-color">
+    <mt-colorpicker
+      class="mt-text-editor-toolbar-button-color__colorpicker"
+      compact
+      :model-value="getTextColor()"
+      @update:modelValue="onUpdateModelValue"
+      :disabled="button.disabled ? button.disabled(props.editor) : disabled"
+      apply-mode
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from "vue";
+import MtColorpicker from "../../mt-colorpicker/mt-colorpicker.vue";
+import type { Editor } from "@tiptap/vue-3";
+import type { CustomButton } from "./mt-text-editor-toolbar.vue";
+import { useI18n } from "vue-i18n";
+
+useI18n({
+  useScope: "global",
+  messages: {
+    en: {
+      "mt-text-editor-toolbar-button-color": {
+        label: "Text Color",
+      },
+    },
+    de: {
+      "mt-text-editor-toolbar-button-color": {
+        label: "Textfarbe",
+      },
+    },
+  },
+});
+
+const props = defineProps({
+  editor: {
+    type: Object as PropType<Editor>,
+    required: true,
+  },
+  button: {
+    type: Object as PropType<CustomButton>,
+    required: true,
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const getTextColor = () => {
+  return props.editor.getAttributes("textStyle").color;
+};
+
+const onUpdateModelValue = (color: string) => {
+  props.editor.chain().focus().setColor(color).run();
+};
+</script>
+
+<script lang="ts">
+export const colorButton: CustomButton = {
+  name: "text-color",
+  label: "mt-text-editor-toolbar-button-color.label",
+  position: 2000,
+};
+</script>
+
+<style lang="scss">
```

Comment:
Would be fantastic if this uses scoped styles and also plain css as we want to get rid of SCSS.